rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/BookKeeperServer.java

```diff
@@ -47,6 +48,7 @@ private BookKeeperServer()
     public static void main(String[] args)
     {
         conf = new Configuration();
+        conf.addResource(new YarnConfiguration());
```

Comment:
If neither master.hostname or yarn property is not present it will default to locahost (which never happens). I had checked with abhishekmodi that both hadoop1 and hadoop2 will always have this property.